src/hotspot/cpu/ppc/assembler_ppc.hpp

```diff
@@ -1245,6 +1305,22 @@ class Assembler : public AbstractAssembler {
   static inline int hi16_signed(  int x) { return (int)(int16_t)(x >> 16); }
   static inline int lo16_unsigned(int x) { return x & 0xffff; }
 
+  static void set_imm18(int* instr, int s) {
+    assert(PowerArchitecturePPC64 >= 10, "Prefixed instructions are supported only in Power 10 and up");
+    // imm18 is in the lower 18 bits of the prefix instruction,
+    // so this is endian-neutral. Same for the get_imm18 below.
+    uint32_t w = *(uint32_t *)instr;
+    *instr = (int)((w & ~0x0003FFFF) | (s & 0x0003FFFF));
+  }
+
+  static int get_imm18(address a, int instruction_number) {
+    assert(PowerArchitecturePPC64 >= 10, "Prefixed instructions are supported only in Power 10 and up");
+    return (((int *)a)[instruction_number] << 14) >> 14;
+  }
+
+  static inline int hi18_signed(  int x) { return hi16_signed(x); }
+  static inline int hi18_signed( long x) { return (int)((x << 30) >> 46); }
```

Comment:
The extra spaces for alignment look a bit strange here.  I think it should be:
```
static inline int hi18_signed( int x) ...
static inline int hi18_signed(long x) ...
```
Basically, remove the leading space from the ( long x) one and delete one from (  int x) to match.